### PR TITLE
Create deltas using API

### PR DESCRIPTION
**Pull request by nadvornik**
For now it still must be run on the server.
This will be changed as soon as there is API for uploading image files.

### CODE REVIEW

File: python-susemanager-retail/retail_create_delta [hunk 1]
```diff
@@ -24,66 +25,47 @@ _cat_list = {
     'tar.xz': 'xzcat'
```
Comment (aaannz) [inline python-susemanager-retail/retail_create_delta:25]:
```python
# OS images path:
MANAGER_PILLAR_DATA_PATH = '/srv/susemanager/pillar_data'
IMAGES_DATA_PATH = os.path.join(MANAGER_PILLAR_DATA_PATH, 'images')
```

above should be deleted, not used anymore

File: python-susemanager-retail/retail_create_delta [hunk 2]
```diff
@@ -24,66 +25,47 @@ _cat_list = {
...
     try:
-        src_bundle = bundle_path(src_pillar.get('sync', {}).get('bundle_url'))
-        target_url = target_pillar.get('sync', {}).get('bundle_url')
-        target_bundle = bundle_path(target_url)
         cat = _cat_list[next(filter(lambda x: target_bundle.endswith(x), _cat_list.keys()))]
 
-        tar = tarfile.open(src_bundle, "r")
+        tar = tarfile.open(img_path + src_bundle, "r")
         tar.extractall(tmpdir)
         tar.close()
```
Comment (aaannz) [inline python-susemanager-retail/retail_create_delta:39]:
```suggestion
        with tarfile.open(img_path + src_bundle, "r") as tar
            tar.extractall(tmpdir)
```

**Review by aaannz — COMMENTED** [review]
I have couple notes. For better UX I would expect ability to print available images, but that is not as important.

**Comment by aaannz** [comment]
Instead of uploading images, can we move this functionality to the java (or salt runner) code in the future development?

**Comment by nadvornik** [comment]
Yes, we can move it to java, but uploading images is needed also for other scenarios.

**Comment by nadvornik** [comment]
Updated according to the suggestions.

**Review by aaannz — APPROVED** [review]
LGTM, I just need to remember from this point this is only for 4.3 and newer.

**Pull request merged**